tests/robustness/validate/watch.go

```diff
@@ -107,14 +110,61 @@ func validateAtomic(t *testing.T, report traffic.ClientReport) {
 	}
 }
 
-func validateReliable(t *testing.T, events []model.WatchEvent) {
-	var lastEventRevision int64 = 1
-	for _, event := range events {
-		if event.Revision > lastEventRevision && event.Revision != lastEventRevision+1 {
-			t.Errorf("Broke watch guarantee: Reliable - a sequence of events will never drop any subsequence of events; if there are events ordered in time as a < b < c, then if the watch receives events a and c, it is guaranteed to receive b, missing revisions from range: %d-%d", lastEventRevision, event.Revision)
+func validateReliable(t *testing.T, events []model.WatchEvent, report traffic.ClientReport) {
+	for _, op := range report.Watch {
+		index := 0
+		revision := firstRevision(op)
+		for index < len(events) && events[index].Revision < revision {
+			index++
+		}
+		if index == len(events) {
+			continue
+		}
+		for _, resp := range op.Responses {
+			for _, event := range resp.Events {
+				if events[index].Match(op.Request) && events[index] != event {
```

Comment:
Do we need to check the index's boundaries ?

Reply:
Technically no, events in report should be a subset of events. This is only test code, so even if someone breaks that assumption we should still be ok with panic.